ros/src/computing/planning/common/lib/astar_search/astar_search/include/astar_search.h

```diff
@@ -0,0 +1,96 @@
+/*
+*/
+
+#ifndef ASTER_PLANNER_H
+#define ASTER_PLANNER_H
+
+#include <iostream>
+#include <vector>
+#include <queue>
+#include <string>
+#include <chrono>
+
+#include <ros/ros.h>
+#include <tf/tf.h>
+#include <nav_msgs/OccupancyGrid.h>
+#include <geometry_msgs/PoseArray.h>
+#include <nav_msgs/Path.h>
+
+#include "astar_util.h"
+
+class AstarSearch
+{
+public:
+  AstarSearch();
+  ~AstarSearch();
+  void initialize(const nav_msgs::OccupancyGrid& costmap);
+  bool makePlan(const geometry_msgs::Pose& start_pose, const geometry_msgs::Pose& goal_pose);
+  void reset();
+
+  nav_msgs::Path getPath() const
```

Comment:
nav_msgs::Path contains vector. It seems better to return its const reference.
```suggestion
  const nav_msgs::Path& getPath() const
```